src/test/cpp/test_unicode.cpp

```diff
@@ -64,10 +64,10 @@ TEST(Unicode, encode_decode_reversible_fuzzing)
     constexpr int iterations = 1'000'000;
 
     std::default_random_engine rng { 12345 };
-    std::uniform_int_distribution<char32_t> distr { 0, code_point_max };
+    std::uniform_int_distribution<std::uint32_t> distr { 0, code_point_max };
 
     for (int i = 0; i < iterations; ++i) {
-        const char32_t code_point = distr(rng);
+        const auto  code_point = static_cast<char32_t>(distr(rng));
```

Comment:
```suggestion
        const auto code_point = char32_t(distr(rng));
```
Please format the code using clang-format before committing. I'll eventually set up a GitHub CI to check this on PRs automatically. For this PR, it would be enough to get rid of the double spaces between `auto` and `code_point`.

I also don't use `static_cast` for conversions between integers, but that's just nitpicking and you can leave it as is if you want to.